### PR TITLE
MWPW-147468 - Delayed Activation of Blade Animations on Ps Product Page

### PR DESCRIPTION
**Pull request by sivasadobe**
* This feature ensures that videos automatically play when they come into view within the viewport and pause when they leave the viewport.
* Added logic to handle video play/pause based on the viewport intersection ratio( ≥0.8 ).

Resolves: [MWPW-147468](https://jira.corp.adobe.com/browse/MWPW-147468)

**Steps for Authors:**

- Edit the video URL to include `#viewportplay`.

### Test URLs

**MILO**
- Before: https://main--milo--sivasadobe.hlx.page/drafts/siva/products/videoplay-onviewport?martech=off
- After: https://mwpw-147468--milo--sivasadobe.hlx.page/drafts/siva/products/videoplay-onviewport?martech=off

**CC**
- Before: https://main--cc--adobecom.hlx.page/drafts/siva/products/photoshop?martech=off
- After: https://main--cc--adobecom.hlx.page/drafts/sivas/products/photoshop?martech=off&milolibs=mwpw-147468--milo--sivasadobe

### CODE REVIEW

File: libs/utils/utils.js [hunk 1]
```diff
@@ -1231,3 +1231,26 @@ export function loadLana(options = {}) {
 }
 
 export const reloadPage = () => window.location.reload();
+
+export function getVideoIntersectionObserver() {
```
Comment (overmyheadandbody) [inline libs/utils/utils.js:1235]: This is probably not a good fit for the utils file, it should be defined closer to where it's needed.

Comment (sivasadobe) [reply libs/utils/utils.js]: @overmyheadandbody @narcis-radu 
Moved `getVideoIntersectionObserver` function to `decorate.js`

File: libs/blocks/video/video.css [hunk 2]
```diff
@@ -5,4 +5,13 @@ a[href*='.mp4'].hide-video {
...
   height: auto;
+  transition: opacity 400ms ease-in-out;
+}
+
+video.in-viewport {
+  opacity: 1;
+}
+
+video.out-viewport {
+  opacity: 0.4;
```
[the remark this answers is not in the document]
Comment (sivasadobe) [reply libs/blocks/video/video.css]: we don't need to apply the opacity animation when `pause` so **removed** this functionality

**Review by aishwaryamathuria — APPROVED** [review]

**Review by suhjainadobe — APPROVED** [review]

**Comment by spadmasa** [comment]
Validation has been done on below urls and details are updated in ticket 
https://mwpw-147468--milo--sivasadobe.hlx.page/drafts/siva/products/videoplay-onviewport?martech=off
https://main--cc--adobecom.hlx.page/drafts/sivas/products/photoshop?martech=off&milolibs=mwpw-147468--milo--sivasadobe
https://jira.corp.adobe.com/browse/MWPW-152904

**Review by drashti1712 — APPROVED** [review]

**Pull request merged**